File: helpers/templates/javascript/benchmark.js
/**
 * Lightweight timing benchmark for {{TITLE}}.
 * Run with: node benchmark.js
 */
// TODO: remove the next line once the loop below calls solve(...)
// eslint-disable-next-line no-unused-vars
const { solve } = require("./solution");

const N = 1000;
const start = process.hrtime.bigint();
for (let i = 0; i < N; i++) {
  // TODO: call solve(...) with real arguments
}
const end = process.hrtime.bigint();
console.log(`${N} iterations: ${Number(end - start) / 1e6} ms`);

File: helpers/templates/cpp/benchmark.cpp
// Lightweight timing benchmark for {{TITLE}}.
#include <chrono>
#include <cstdio>

#include "solution.h"

int main() {
    const int n = 1000;
    auto start = std::chrono::high_resolution_clock::now();
    for (int i = 0; i < n; i++) {
        solve();
    }
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double, std::milli> ms = end - start;
    std::printf("%d iterations: %.3f ms\n", n, ms.count());
    return 0;
}
